contracts/HubPool.sol

```diff
@@ -339,19 +459,35 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         return _liquidityUtilizationPostRelay(l1Token, relayedAmount);
     }
 
-    function sync(address l1Token) public nonReentrant {
+    /**
+     * @notice Synchronize any balance changes in this contract with the utilized & liquid reserves. This should be done
+     * at the conclusion of an L2 -> L1 token transfer via the canonical token bridge, for example, when this contract's
+     * reserves do not reflect all of its token balance.
+     */
+    function sync(address l1Token) public override nonReentrant {
         _sync(l1Token);
     }
 
     /*************************************************
      *             DATA WORKER FUNCTIONS             *
      *************************************************/
 
-    // After proposeRootBundle is called, if the any props are wrong then this proposal can be challenged. Once the
-    // challenge period passes, then the roots are no longer disputable, and only executeRootBundle can be called and
-    // this can't be called again until all leafs are executed.
-    // Note: bundleEvaluationBlockNumbers must contain the latest block number for _all_ chains, even if there are no
-    // relays contained on some of them.
+    /**
+     * @notice Publish a new root bundle to this contract along with all of the block numbers that the merkle roots
+     * are relevant for. This is used to aid off-chain validators in evaluating the correctness of this bundle. Caller
+     * must stake a bond that can be slashed if the root bundle proposal is invalid, and they will receive it back
+     * if accepted.
+     * @notice After proposeRootBundle is called, if the any props are wrong then this proposal can be challenged.
+     * Once the challenge period passes, then the roots are no longer disputable, and only executeRootBundle can be
+     * called; moreover, this method can't be called again until all leafs are executed.
+     * @param bundleEvaluationBlockNumbers should contain the latest block number for _all_ chains, even if there are no
+     * relays contained on some of them. But the usage of this variable should be defined in an off chain UMIP.
+     * @param poolRebalanceLeafCount Number of leaves contained in pool rebalance root.
+     * @param relayerRefundRoot Relayer refund root to publish to Spoke Pool where a data worker can execute leaves to
+     * refund relayers.
+     * @param slowRelayRoot Slow relay root to publish to Spoke Pool where a data worker can execute leaves to
+     * fulfill slow relays.
+     */
```

Comment:
you are missing a comment on `poolRebalanceRoot`

Reply:
done